packages/core/admin/admin/src/content-manager/hooks/useRelation/tests/useRelation.test.js

```diff
@@ -107,6 +107,49 @@ describe('useRelation', () => {
     expect(result.current.relations.data.pages[0]?.data).toBeUndefined();
   });
 
+  test('fetch relations, reverse inner pages and outer pages', async () => {
```

Comment:
I find the test a little confusing, aren't we just wanting to make sure that the relations are fetched and reversed in which case we could possibly reduce this to just checking `FIRST_PAGE_FIXTURE`. Not sure I see the importance of adding another page? Happy to be told i'm wrong though! 😄 

Another side note, it might be better not to use `waitForNextUpdate` because when we move to `react18` we will need to remove the usage of `@testing-library/react-hooks` and move to `@testing-library/react` which iirc does not have this function.

We could do `waitFor(() => result.current.relations.data.page.length > 0)` (something like that) which will move over easier. Just a thought.

Reply:
ah good to know for `waitForNextUpdate` I will check if I can make it work without it!

the double page calling is because I want to test that it reverses the elements inside the page itself so:
```js
// what we get
pages-1: [{id: 1}, {id:2}] 

// what we want
pages-1: [{id: 2}, {id:1}]
```

but also that it reverses the pages themselves when we get multiple pages:
```js
// what we get
pages-1: [{id: 1}, {id:2}]
pages-2: [{id: 3}, {id:4}]

// what we want
pages-1: [{id: 4}, {id:3}]
pages-2: [{id: 2}, {id:1}]
```

since this double reverse logic is a bit confusing I thought it would be worth it to test it, make sure it works + explanation of what is happening
but I guess I can remove the first `expect` since both reverse can be tested on the second call 🤔
I can also add a comment in the `useRelation` `select` for more documentation
wdyt?

Reply:
I think if you've got a valid reason (which sounds like you have) then it's totally fine to implement this way. I would write an explanation or make your test name longer in the code so others are clear of the intentions!

I would assert that the first page works correctly again after the first update to be confident it works on initial fetching and nothing odd happens in between 💪🏼

Reply:
yeah...... close it 👀 probs easier for everyone.